### PR TITLE
Add uncertainty quantification to the `predict` CLI

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
Make uncertainty prediction, calibration, and evaluations available via the CLI. This PR will also add CLI tests and documentation. Thus, this PR will be merged after the other PRs related to uncertainty implementations are completed.

## Questions
I am considering not supporting `--evaluation-scores-path` and instead just printing the evaluation scores in the log.

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by shihchengli** [comment]
This PR depends on #1072 and #1070

File: chemprop/cli/predict.py [hunk 1]
```diff
@@ -183,110 +227,131 @@ def make_prediction_for_models(
...
 
     df_test = pd.read_csv(
         args.test_path, header=None if args.no_header_row else "infer", index_col=False
     )
-    df_test[output_columns] = average_preds
+    df_test[output_columns] = test_preds
+
+    if args.uncertainty_method not in ["none", "classification"]:
+        unc_columns = [f"{col}_unc" for col in output_columns]
+        df_test[unc_columns] = test_uncs
```
Comment (KnathanM) [inline chemprop/cli/predict.py:354]: I suggest printing less decimal places than is the default. Maybe 6 at most. Right now it gives numbers like `1.7383126678770482`.

File: chemprop/cli/predict.py [hunk 2]
```diff
@@ -183,110 +227,131 @@ def make_prediction_for_models(
...
-    ]
-
-    if multicomponent:
-        test_dset = data.MulticomponentDataset(test_dsets)
+    test_individual_preds, test_individual_uncs = uncertinaty_predictor(
+        test_loader, models, trainer
+    )
+    test_preds = torch.mean(test_individual_preds, dim=0)
+    if not isinstance(uncertinaty_predictor, NoUncertaintyPredictor):
+        test_uncs = torch.mean(test_individual_uncs, dim=0)
```
[the remark this answers is not in the document]
Comment (shihchengli) [reply chemprop/cli/predict.py:258]: I think it is different to apply calibration to the predicted uncertainty for each model compared to the average predicted uncertainty across ensemble models. For example, the scaler obtained from maximizing NLL between the individual predictions and the targets would differ from that of the average predictions and the targets. However, I can also see that the averaged of the calibrated uncertainty across ensemble models is not the same as the average calibrated uncertainty confusing.

File: chemprop/cli/predict.py (hunk 1 above)
Comment (shihchengli) [reply chemprop/cli/predict.py:354]: Good idea

File: chemprop/cli/predict.py [hunk 3]
```diff
@@ -183,110 +227,131 @@ def make_prediction_for_models(
...
+        test_uncs = None
 
-    if isinstance(model.predictor, MveFFN) or isinstance(model.predictor, EvidentialFFN):
-        average_preds = average_preds[..., 0]
+    if args.calibration_method is not None:
+        if args.calibration_method == "platt":
+            raise NotImplementedError(
+                "`PlattCalibrator` requires the number of positive and negative training examples "
+                "to adjust the target probability values. This method is not currently exposed in the predict CLI."
+            )
```
[the remark this answers is not in the document]
Comment (shihchengli) [reply chemprop/cli/predict.py:267]: Good point! I will update it.

File: chemprop/cli/predict.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shihchengli) [reply chemprop/cli/predict.py:258]: As we are using the same data split for ensemble models, I think it doesn't hurt to fit the calibrator based on the average predicted uncertainty and apply it to each individual prediction.

**Review by KnathanM — APPROVED** [review]

**Review by KnathanM — APPROVED** [review]

**Pull request merged**